tox.ini

```
@@ -55,6 +55,14 @@ commands =
     mypy --python-version 3.5 src/build
     mypy --python-version 2.7 src/build
 
+[testenv:{py39, py38, py37, py36, py35, py27, pypy3, pypy2}-minversion]
+description = check minimum versions required of all dependencies
+extras =
+    test
+commands =
+    pip install . -r tests/requirements-min.txt
```

We could first build a wheel and then perform the install here of that? Rather than installing the folder itself.

---

That's not how you use a constraints file:

```suggestion
    pip install -c tests/requirements-min.txt .
```

---

Constraints do not install a package listed if not listed anywhere else (in pip 20.3+), but that's all. The solver still includes the combined requirements. I've verified that a mismatch does fail. The -c is better, since this is exactly a constraint file and shouldn't install anything, but it still works as a requirements file.

---

We don't pre-build a wheel for any of the other tests?

---

We do within the suite. But probably you don't want to use that here.